kep/594-resourcepolicy/README.md

```diff
@@ -0,0 +1,191 @@
+- [Known limitations](#known-limitations)
+- [Test plans](#test-plans)
+- [Graduation criteria](#graduation-criteria)
+- [Feature enablement and rollback](#feature-enablement-and-rollback)
+<!-- /toc -->
+
+## Summary
+This proposal introduces a plugin to allow users to specify the priority of different resources and max resource 
+consumption for workload on differnet resources.
+
+## Motivation
+The machines in a Kubernetes cluster are typically heterogeneous, with varying CPU, memory, GPU, and pricing. To 
+efficiently utilize the different resources available in the cluster, users can set priorities for machines of different 
+types and configure resource allocations for different workloads. Additionally, they may choose to delete pods running 
+on low priority nodes instead of high priority ones. 
+
+### Use Cases
+
+1. As a user of cloud services, there are some stable but expensive ECS instances and some unstable but cheaper Spot 
+instances in my cluster. I hope that my workload can be deployed first on stable ECS instances, and during business peak 
+periods, the Pods that are scaled out are deployed on Spot instances. At the end of the business peak, the Pods on Spot 
+instances are prioritized to be scaled in.
+
+### Goals
+
+1. Develop a filter plugin to restrict the resource consumption on each unit for different workloads.
+2. Develop a score plugin to favor nodes matched by a high priority unit.
+3. Automatically setting deletion costs on Pods to control the scaling in sequence of workloads through a controller.
+
+### Non-Goals
+
+1. Modify the workload controller to support deletion costs. If the workload don't support deletion costs, scaling in 
+sequence will be random.
+2. When creating a ResourcePolicy, if the number of Pods has already violated the quantity constraint of the 
+ResourcePolicy, we will not attempt to delete the excess Pods.
+
+
+## Proposal
+
+### CRD API
+```yaml
+apiVersion: scheduling.sigs.x-k8s.io/v1alpha1
+kind: ResourcePolicy
+metadata:
+  name: xxx
+  namespace: xxx
+spec:
+  matchLabelKeys:
+    - pod-template-hash
+  matchPolicy:
+    ignoreTerminatingPod: true
+    ignorePreviousPod: false
+  forceMaxNum: false
+  podSelector:
+    matchExpressions:
+      - key: key1
+        operator: In
+        values:
+        - value1
+    matchLabels:
+      key1: value1
+  strategy: prefer
+  units:
+  - name: unit1
+    priority: 5
+    maxCount: 10
+    nodeSelector:
+      matchExpressions:
+      - key: key1
+        operator: In
+        values:
+        - value1
+  - name: unit2
+    priority: 5
+    maxCount: 10
+    nodeSelector:
+      matchExpressions:
+      - key: key1
+        operator: In
+        values:
+        - value2
+  - name: unit3
+    priority: 4
+    maxCount: 20
+    nodeSelector:
+      matchLabels:
+        key1: value3
+```
+
+`Priority` define the priority of each unit. Pods will be scheduled on units with a higher priority. 
```

Comment:
is `Prirority` is under `units[*]`? if so, it'd read better to use a bullet to explain it under a top-level `units` API field.

Reply:
I think we missed a detailed explanation for `units`.

Reply:
Updated the kep. Thanks